ingestion/src/metadata/ingestion/source/messaging/kinesis/metadata.py

```diff
@@ -162,19 +168,20 @@ def _get_topic_partitions(self, topic_name: str) -> List[str]:
         all_partitions, has_more_partitions, args = (
             [],
             True,
-            {KinesisEnum.STREAM_NAME.value: topic_name},
+            KinesisStreamArgs(StreamName=topic_name),
         )
         try:
             while has_more_partitions:
-                partitions = self.kinesis.list_shards(**args)
+                partitions = self.kinesis.list_shards(**args.dict())
+                kinesis_partitions_model = KinesisPartitions(**partitions)
                 all_partitions.extend(
                     [
-                        partition.get(KinesisEnum.SHARD_ID.value)
-                        for partition in partitions.get(KinesisEnum.SHARDS.value) or []
+                        partition.ShardId
+                        for partition in kinesis_partitions_model.Shards or []
                     ]
                 )
-                has_more_partitions = partitions.get(KinesisEnum.NEXT_TOKEN.value)
-                args[KinesisEnum.NEXT_TOKEN.value] = has_more_partitions
+                has_more_partitions = kinesis_partitions_model.NextToken
+                args.NextToken = has_more_partitions
```

Comment:
Why `NextToken` is not a property of `KinesisArgs`?

Reply:
Do we need `has_more_partitions` if we change this into a do-while and we do this assignation `args.NextToken = kinesis_partitions_model.NextToken`?

Reply:
For the `NextToken`, same as the reason [here](https://github.com/open-metadata/OpenMetadata/pull/10578#discussion_r1142495077)

Reply:
Python does not support do-while loops so we might not be able to do this

> Do we need `has_more_partitions` if we change this into a do-while and we do this assignation `args.NextToken = kinesis_partitions_model.NextToken`?